### PR TITLE
fix(@aws-amplify/ui-components): formFields hint handle

### PR DESCRIPTION
**Pull request by ashika01**
_Issue #, if available:_ #5298

_Description of changes:_
Handle provide hint with default value under certain conditions.

This PR handles hint for password and code and  in the following ways,
- `hint: null` is passed to `formFields`
image

- string is to `hint` in `formFields`
image

- if nothing is passed in or a function component, it would resolve to fallback
image

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ericclemmons** [comment]
Can you share a screenshot/gif of how this looks when used? As well as how, if at all, customers would customize hints?  

I couldn't make it out from the code alone 😅

**Comment by jordanranz** [comment]
Can we put this on `amplify-auth-fields` instead of individual components

**Comment by jordanranz** [comment]
Looking at it further, I am not sure my previous comment would work since only certain subcomponents have hints. Maybe we should allow for a hint on every form field used.

**Comment by ashika01** [comment]
> Looking at it further, I am not sure my previous comment would work since only certain subcomponents have hints. Maybe we should allow for a hint on every form field used.

@jordanranz That might be the way to go and have it `null` by default unless something is mentioned.

I am probably gonna redo hint after we merge this PR https://github.com/aws-amplify/amplify-js/pull/5691

**Comment by ashika01** [comment]
@jordanranz After looking into this a little bit, I think this is probably the best approach for unblocking our customers for now. And like we discussed, we should do api design around FormFields and think about this?!

**Comment by ashika01** [comment]
@ericclemmons @jordanranz Updated the description with whats happening right now. Let me know your thoughts.?

File: packages/amplify-ui-components/src/common/helpers.ts [hunk 1]
```diff
@@ -63,3 +63,12 @@ export const onAuthUIStateChange = (authStateHandler: AuthStateHandler) => {
   Hub.listen(UI_AUTH_CHANNEL, authUIStateHandler);
   return () => Hub.remove(UI_AUTH_CHANNEL, authUIStateHandler);
 }
+
+export const handleAuthFormFieldHint = (field) => {
```
Comment (jordanranz) [inline packages/amplify-ui-components/src/common/helpers.ts:67]: can we change this to `isHintValid` and just have the content be:
```
return !(field["hint"] === null || typeof field["hint"] === "string")
```

**Pull request merged**